sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -537,6 +537,18 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog with QueryErrorsB
               }
             }
 
+          case Window(_, partitionSpec, _, _) =>
+            // Both `partitionSpec` and `orderSpec` must be orderable. We only need an extra check
+            // for `partitionSpec` here because `orderSpec` has the type check itself.
+            partitionSpec.foreach { p =>
+              if (!RowOrdering.isOrderable(p.dataType)) {
+                p.dataTypeMismatch(p, TypeCheckResult.DataTypeMismatch(
+                  errorSubClass = "INVALID_ORDERING_TYPE",
```

Comment:
why not use the error class `EXPRESSION_TYPE_IS_NOT_ORDERABLE`?

Reply:
I cannot really tell the difference between these two error classes, so I just randomly picked one. Now I changed it into `EXPRESSION_TYPE_IS_NOT_ORDERABLE` instead.